pkg/kv/kvserver/replica_raft.go

```diff
@@ -833,21 +833,24 @@ func (r *Replica) handleRaftReadyRaftMuLocked(
 		level := kvflowcontrol.GetV2EnabledWhenLeaderLevel(
 			ctx, r.store.ClusterSettings(), r.store.TestingKnobs().FlowControlTestingKnobs)
 		if level > r.raftMu.flowControlLevel {
-			if r.raftMu.flowControlLevel == kvflowcontrol.V2NotEnabledWhenLeader {
-				func() {
-					r.mu.Lock()
-					defer r.mu.Unlock()
+			var basicState replica_rac2.RaftNodeBasicState
+			func() {
+				r.mu.Lock()
+				defer r.mu.Unlock()
+				basicState = replica_rac2.MakeRaftNodeBasicStateLocked(
+					r.mu.internalRaftGroup, r.shMu.state.Lease.Replica.ReplicaID)
```

Comment:
Why is this not part of the same `Replica.mu` transaction with the `Ready`-getting below? Is it possible to avoid doing this work twice and also needing to think about what happens when the two sections get out sync?